pyrobosim_ros/pyrobosim_ros/ros_interface.py

```diff
@@ -535,6 +553,33 @@ def robot_path_cancel_callback(self, goal_handle, robot=None):
         self.get_logger().info(f"Canceling path following for {robot}.")
         return CancelResponse.ACCEPT
 
+    def robot_path_planner_reset_callback(self, request, response, robot=None):
+        """
+        Resets a robot's path planner as a response to a service request.
+
+        :param request: The service request.
+        :type request: :class:`std_srvs.srv.Trigger.Request`
+        :param response: The unmodified service response.
+        :type response: :class:`std_srvs.srv.Trigger.Response`
+        :return: The modified service response containing the reset result.
+        :rtype: :class:`std_srvs.srv.Trigger.Response`
+        """
+        self.get_logger().info(f"Resetting path planner for {robot}.")
+
+        if not robot.path_planner:
+            message = f"{robot} does not have a path planner. Cannot reset."
+            self.get_logger().warn(message)
+            return Trigger.Response(success=False, message=message)
+
+        robot.path_planner.reset()
```

Comment:
Wait, so is the reset function mandatory? The docs make it sound optional.

Reply:
It is mandatory. I will clear up the docs.